minio/minio.go

```diff
@@ -0,0 +1,170 @@
+package minio
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// Storage interface that is implemented by storage providers
+type Storage struct {
+	minio *minio.Client
+	cfg   Config
+	ctx   context.Context
+}
+
+// New creates a new storage
+func New(config ...Config) *Storage {
+
+	// Set default config
+	cfg := configDefault(config...)
+
+	// Minio instance
+	minioClient, err := minio.New(cfg.Endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(cfg.Credentials.accessKeyID, cfg.Credentials.secretAccessKey, cfg.Token),
+		Secure: cfg.Secure,
+		Region: cfg.Region,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	storage := &Storage{minio: minioClient, cfg: cfg, ctx: context.Background()}
+
+	// Reset all entries if set to true
+	if cfg.Reset {
+		if err = storage.Reset(); err != nil {
+			panic(err)
+		}
+	}
+
+	// check bucket
+	err = storage.CheckBucket()
+	if err != nil {
+		// create bucket
+		err = storage.CreateBucket()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return storage
+}
+
+// Get value by key
+func (s *Storage) Get(key string) ([]byte, error) {
+
+	if len(key) <= 0 {
+		return nil, errors.New("the key value is required")
+	}
+
+	// get object
+	object, err := s.minio.GetObject(s.ctx, s.cfg.Bucket, key, s.cfg.GetObjectOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert to byte
+	buf := new(bytes.Buffer)
```

Comment:
Can we use bytebufferpool here? It's more performant

Reply:
done